File: shared/src/commonMain/kotlin/com/example/plauenblod/feature/auth/repository/AuthRepository.kt
package com.example.plauenblod.feature.auth.repository

import com.example.plauenblod.feature.auth.AuthResult
import com.example.plauenblod.feature.auth.UserRole
import kotlinx.coroutines.flow.StateFlow

interface AuthRepository {
    val isLoggedIn: StateFlow<Boolean>
    val isInitialized: StateFlow<Boolean>

    suspend fun signUp(userName: String, email: String, password: String): AuthResult
    suspend fun addUserToFirestore(userId: String, userName: String, role: UserRole = UserRole.USER)
    suspend fun signIn(email: String, password: String): AuthResult
    suspend fun fetchUserRole(userId: String): UserRole?
    suspend fun sendPasswort(email: String): Result<Unit>
    fun signOut()
}

File: Bouldery/feature/authentication/repository/AuthRepository.swift
//
//  AuthRepository.swift
//  Bouldery
//
//  Created by Kim Reuter on 20.08.25.
//

import Foundation
import Combine

protocol AuthRepository: AnyObject {
    
    var isLoggedInPublisher: AnyPublisher<Bool, Never> { get }
    var isInitializedPublisher: AnyPublisher<Bool, Never> { get }

    var isLoggedInValue: Bool { get }
    var isInitializedValue: Bool { get }

    func signUp(userName: String, email: String, password: String) async -> AuthResult
    func addUserToFirestore(userId: String, userName: String, role: UserRole) async
    func signIn(email: String, password: String) async -> AuthResult
    func fetchUserRole(userId: String) async -> UserRole?
    func sendPasswort(email: String) async -> Result<Void, Error>
    func getCurrentUserId() -> String?
    func signOut()
    
}
